### ISSUE TITLE
Allow to use internal repository to resolve idea artifacts

### ISSUE THREAD

**Issue by slavashvets** [xtext/xtext-gradle-plugin#51]
It would be nice if we were able to download artifacts from internal repository (mirror) during "downloadIdea" and "downloadPlugins" tasks.

For example:

```
archiveUrl = "http://company.com/repos/jetbrains/com/jetbrains/intellij/idea/ideaIC/ideaIC-${ideaVersion}.zip"
explicitPlugins {
  plugin('org.eclipse.xtend.idea', "http://company.com/repos/xtext/idea/${xtextVersion}/org.eclipse.xtend.idea-${xtextVersion}.zip")
  plugin('org.eclipse.xtext.idea', "http://company.com/repos/xtext/idea/${xtextVersion}/org.eclipse.xtext.idea-${xtextVersion}.zip")
  plugin('org.eclipse.xtext.xtext.idea', "http://company.com/repos/xtext/idea/${xtextVersion}/org.eclipse.xtext.xtext.idea-${xtextVersion}.zip")
}
```

**Comment by oehme** [comment]
I agree about the IDEA zip. Would you be interested in contributing a fix? The expected DSL would be

```
ideaDevelopment {
  jetbrainsRepository = "http://company.com/repos/jetbrains"
}
```

For the plugins, you already control the repositories, so I don't understand that request.

**Comment by slavashvets** [comment]
> For the plugins, you already control the repositories, so I don't understand that request.

In the current approach it looks like every time I update xtext version I need to manually deploy `updatePlugins.xml` to the internal repository. Otherwise, if I use cached copy of `updatePlugins.xml` from proxy (mirror) repository, inside it still is a reference to the `download.eclipse.org`.

```
    pluginRepositories {
        url "http://mymirror/${xtextVersion}/updatePlugins.xml"
    }
```
This config will download zip files from `download.eclipse.org`, while I want to download from `http://mymirror`.

**Comment by oehme** [comment]
That is unfortunately a limitation of the updatePlugins.xml format. I'm not a big fan of it myself ;)

What we could do is allowing to specify repositories that don't end in "updatePlugins.xml" and just trying to get the artifacts from there and handling "not found" gracefully. Would that help you?

**Comment by slavashvets** [comment]
How it should look like in DSL?

**Comment by oehme** [comment]
Same as we already have:

```
pluginRepositories {
    url "http://mymirror/${xtextVersion}"
}
```

The difference is just that there is no upatePlugins.xml in that String.

**Comment by slavashvets** [comment]
In this case, where it takes a list of IDs? Is it just predefined?

**Comment by oehme** [comment]
The plugin IDs are specified in the `pluginDependencies` section. The name of the zip file can be directly determined from the plugin ID.

**Comment by slavashvets** [comment]
Thanks. I forgot about this.

**Comment by oehme** [comment]
Any update on this? :)

**Comment by Treverix** [comment]
I'm interested in a solution too, because now, after the 2.11 release, the idea plugin subproject for an xtext project is obviously broken because the URL in the build.gradle points to a non-existing plugin repository (http://download.eclipse.org/modeling/tmf/xtext/idea/2.11.0/updatePlugins.xml)

I can build xtext-idea locally (and run the entities example on the 2.11 maintenance branch) and it would be an asset to know how to change the gradle build file in  the project so that I can use the plugin from my local filesystem.

**Comment by oehme** [comment]
@spoenemann ☝️ this should be fixed. Creating an idea project should just work.

@Treverix You can change that url to anything you want, including a file: URL.

**Comment by spoenemann** [comment]
> this should be fixed. Creating an idea project should just work.

Yeah, we need to change that to the last released version of xtext-idea (2.10.0), since we haven't released it for 2.11.0 (noone seemed to be interested in maintaining it).

**Comment by Treverix** [comment]
Thank you for the swift replies! I was successful :)

First I grabbed the 2.10.0 `updatePlugins.xml` from the http url and stored it as a new file. On that file I replaced the `http://` url for the xtext-idea plugin with a `file://` url that pointed to my `2.11.1-SNAPSHOT` plugin.

Then I replaced the `ideaDevelopment` in `build.gradle` with something like this:

```
ideaDevelopment {
	ideaVersion = 'LATEST-EAP-SNAPSHOT'
	pluginRepositories {
		url "file:///<path-to>/xtext-idea/updatePlugins.xml"
	}
	pluginDependencies {
		id 'org.eclipse.xtext.idea' version xtextVersion
	}
}

```

That did it. Built the mydsl xtext artifact, executed runIdea and I had an IDEA with 2.11.1 running inside, started from within the eclipse dev environment. 

Only that the .idea plugin has error markers on each and every generated source file (after generating the xtext artifact. So something is still missing there.
